File: ast/src/testing/vanila/ruby_minimal/app/controllers/post_controller.rb
require_relative '../../lib/models'
require_relative '../../lib/utils'

class PostController
  def self.index
    posts = Models::Post.all.map(&:to_h)
    Utils.json_response(posts)
  end

  def self.create(params)
    post = Models::Post.new(params)
    if post.save
      Utils.json_response(post.to_h, 201)
    else
      Utils.json_response({ error: 'Invalid post data' }, 422)
    end
  end
end

File: ast/src/testing/vanila/php_minimal/controllers/PostController.php
<?php

require_once __DIR__ . '/../models/Post.php';
require_once __DIR__ . '/../includes/helpers.php';

class PostController {
    public function index() {
        $posts = Post::all();
        jsonResponse($posts);
    }

    public function create() {
        $data = json_decode(file_get_contents('php://input'), true);
        
        $post = new Post(
            sanitize($data['title']), 
            sanitize($data['content']), 
            $data['user_id']
        );
        $post->save();
        
        jsonResponse(['id' => $post->id, 'message' => 'Post created'], 201);
    }
}
